### PR TITLE
Update to latest version of colonyNetwork contracts

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

This pull request updates colonyJS to the following version of the colonyNetwork contracts:

`5acd5e2526ffdd9b9577b340f9c8dcf3c22df5ce`

See the [changelog](https://github.com/JoinColony/colonyJS/blob/maintenance/latest-contracts/CHANGELOG.md) for a working list of updates.

## Checklist

- [x] Update and improve client constants
- [x] Update ColonyClient methods and events
- [x] Update ColonyNetworkClient methods and events
- [x] Manually test and apply any necessary fixes

## Notable Changes

- Updated caller methods that retrieve an address to all follow a consistent format. "Address" has been specified in the method name (`getAuthorityAddress`, `getOwnerAddress`, `getTokenAddress`, `getColonyNetworkAddress`) and each returns `address` as an output parameter.

- Updated sender methods that set colony roles (`setFounderRole`, `setAdminRole`, `setRootRole`, `setAdministrationRole`, `setArchitectureRole`, and `setFundingRole`) to use `address` as an input parameter rather than `user` because the colony role will not always be assigned to a "user" address and could be assigned to a contract address (for example, the `OneTxPayment` contract).

- Updated events associated with setting colony roles (`ColonyAdministrationRoleSet`, `ColonyArchitectureRoleSet`, `ColonyFundingRoleSet`, `ColonyRootRoleSet`) to use `address` as a parameter rather than `user` because the colony role will not always be assigned to a "user" address and could be assigned to a contract address (for example, the `OneTxPayment` contract).

- Updated `hasUserRole` to `hasColonyRole` because the role assignment will not always be a "user" address and could be a contract address (for example, the `OneTxPayment` contract). Also, changed the `user` input parameter to `address`.

- Updated the `users` input parameter to `addresses` in the `bootstrapColony` sender method.

- Updated sender methods that set task roles (`setTaskManagerRole`, `setTaskEvaluatorRole`, `setTaskWorkerRole`) to use `address` as an input parameter rather than `user`.

- Updated `makeTask` sender method to `addTask` to be consistent with the other `add` methods (`addDomain`, `addPayment`, `addSkill`, etc.) while keeping `makePayment`, which is associated with the `OneTxPayment` method.

## Next Steps

- [ ] Automatically determine method permission proofs (#386)
- [ ] Update example methods throughout documentation
- [ ] Finish #308 to ensure methods are tested before publishing

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,145 @@
...
+  * Added `getMiningResolverAddress`
+  * Added `getReplacementReputationUpdateLogEntry`
+  * Added `getReputationMiningCycle`
+  * Added `getReputationMiningSkillId`
+  * Added `getReputationRootHash`
+  * Added `getReputationRootHashNodeCount`
+  * Added `getTokenLockingAddress`
+  * Removed `getRootGlobalSkillId` (See `getReputationMiningSkillId`)
+  * Removed `getTokenLocking` (See `getTokenLockingAddress`)
+  * Updated `getSkill`
```
Comment (JamesLefrere) [inline CHANGELOG.md]: Can we add a bullet under this explaining the change?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,145 @@
...
+  * Added `ColonyRootRoleSet`
+  * Added `PaymentAdded`
+  * Added `PayoutClaimed`
+  * Removed `ColonyAdminRoleRemoved` (See `ColonyAdministrationRoleSet`)
+  * Removed `ColonyAdminRoleSet` (See `ColonyAdministrationRoleSet`)
+  * Removed `ColonyFounderRoleSet` (See `ColonyRootRoleSet`)
+  * Removed `TaskPayoutClaimed` (See `PayoutClaimed`)
+  * Updated `ColonyBootstrapped`
+    * Changed `users` parameter to `addresses`
+  * Updated `ColonyInitialised`
```
Comment (JamesLefrere) [inline CHANGELOG.md]: Can we add a bullet under this explaining the change?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,145 @@
...
+  * Added `deprecateSkill`
+  * Added `initialiseReputationMining`
+  * Added `registerColonyLabel`
+  * Added `setMiningResolver`
+  * Added `setReplacementReputationUpdateLogEntry`
+  * Added `setReputationRootHash`
+  * Added `startNextMiningCycle`
+  * Updated `addSkill`
+    * Removed `globalSkill` input parameter
+  * Updated `startTokenAuction`
```
Comment (JamesLefrere) [inline CHANGELOG.md]: Can we add a bullet under this explaining the change?

File: packages/colony-js-client/src/ColonyClient/callers/GetTask.js [hunk 4]
```diff
@@ -39,9 +39,8 @@ export default class GetTask extends ContractClient.Caller<
         ['deliverableHash', 'ipfsHash'],
         ['status', 'taskStatus'],
         ['dueDate', 'date'],
-        ['payoutsWeCannotMake', 'number'],
         ['potId', 'number'],
-        ['completionDate', 'date'],
+        ['completionTimestamp', 'date'],
```
Comment (JamesLefrere) [inline packages/colony-js-client/src/ColonyClient/callers/GetTask.js]: What's the rationale for this change? Aren't we returning a `Date` rather than a number?

File: CHANGELOG.md (hunk 2 above)
Comment (ryanchristo) [reply CHANGELOG.md]: Yes, can do. I was planning to polish up the changelog.

File: packages/colony-js-client/src/ColonyClient/callers/GetTask.js (hunk 4 above)
Comment (ryanchristo) [reply packages/colony-js-client/src/ColonyClient/callers/GetTask.js]: colonyNetwork is using `completionTimestamp` but we can continue using `completionDate`.

File: packages/colony-js-client/src/ColonyClient/callers/GetExtensionAddress.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+import ContractClient from '@colony/colony-js-contract-client';
+
+import type ColonyClient from '../index';
+
+type InputValues = {
+  contractName: string,
+};
+
+export default class GetExtensionAddress extends ContractClient.Caller<
```
Comment (sprusr) [inline packages/colony-js-client/src/ColonyClient/callers/GetExtensionAddress.js]: I think a comment explaining what this is for would help.

File: packages/colony-js-client/src/ColonyClient/senders/AddExtension.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+import ContractClient from '@colony/colony-js-contract-client';
+
+import type ColonyClient from '../index';
+
+type InputValues = {
+  contractName: string,
+};
+
+export default class AddExtension extends ContractClient.Sender<
```
Comment (sprusr) [inline packages/colony-js-client/src/ColonyClient/senders/AddExtension.js]: Hm maybe it would be helpful to have an `ExtensionFactoryClient`. Do you think that could make this simpler? (I'm suggesting this without really thinking too much into it, so it could be an awful idea!)

File: packages/colony-js-client/src/ColonyClient/senders/RemoveExtension.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+import ContractClient from '@colony/colony-js-contract-client';
+
+import type ColonyClient from '../index';
+
+type InputValues = {
+  contractName: string,
+};
+
+export default class RemveExtension extends ContractClient.Sender<
```
Comment (sprusr) [inline packages/colony-js-client/src/ColonyClient/senders/RemoveExtension.js]: Again, could this be in an `ExtensionFactoryClient`? Also typo!

File: packages/colony-js-client/src/ColonyClient/senders/SetAdminRole.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+import type ColonyClient from '../index';
+
+type Address = string;
+
+type InputValues = {
+  address: Address,
+  setTo: boolean,
+};
+
+export default class OneTxPayment extends ContractClient.Sender<
```
Comment (sprusr) [inline packages/colony-js-client/src/ColonyClient/senders/SetAdminRole.js]: I think this class should be named something else :)

File: packages/colony-js-client/src/ColonyClient/senders/AddExtension.js (hunk 6 above)
Comment (ryanchristo) [reply packages/colony-js-client/src/ColonyClient/senders/AddExtension.js]: With only three methods related to what might be an `ExtensionsFactoryClient` (deploy, remove, get address) and because extension contracts are closely tied to a colony, I think it would be better to bake them in to the colony client (at least for now) but maybe a separate client would be more appropriate in the future.
